js/src/block-editor/hooks/useMedia.js

```diff
@@ -50,14 +56,21 @@ import { __, sprintf } from '@wordpress/i18n';
  */
 const useMedia = ( fieldValue, onChange, allowedTypes ) => {
 	const defaultImageSrc = '';
+	const [ media, setMedia ] = useState( { alt: undefined, source_url: undefined } );
 	const [ mediaSrc, setMediaSrc ] = useState( defaultImageSrc );
 	const [ isUploading, setIsUploading ] = useState( false );
 	const [ mediaAlt, setImageAlt ] = useState( '' );
 
-	/* @type {Object|undefined} */
-	const media = useSelect( ( select ) => {
-		// @ts-ignore The function isn't in the declaration file.
-		return select( 'core' ).getMedia( fieldValue );
+	const getImage = useSelect( ( select ) => {
+		/**
+		 * Gets the image.
+		 *
+		 * @param {number | string} imageId The id of the image.
+		 * @return {GetImageReturn | undefined} The image, if any.
+		 */
+		return ( imageId ) =>
+			// @ts-ignore This function is wrong in the declaration file.
+			select( 'core' ).getEntityRecord( 'postType', 'attachment', imageId, { context: 'embed' } );
```

Comment:
`select( 'core' ).getMedia( fieldValue )` sometimes failed, like when an Author requests an image that an Admin added.

So this uses another type of request. It's hacky also, but I couldn't find another way to fix this.